Kernel/Tasks/PowerManagementTask.cpp

```diff
@@ -170,36 +170,8 @@ void PowerManagementTask::perform_shutdown(DoReboot do_reboot)
 
     dbgln("Unmounting all file systems...");
 
-    auto unmount_was_successful = true;
-    while (unmount_was_successful) {
-        unmount_was_successful = false;
-        Vector<Mount&, 16> mounts;
-        MUST(VirtualFileSystem::the().for_each_mount([&](auto const& mount) -> ErrorOr<void> {
-            MUST(mounts.try_append(const_cast<Mount&>(mount)));
-            return {};
-        }));
-        if (mounts.is_empty())
-            break;
-        auto const remaining_mounts = mounts.size();
-
-        while (!mounts.is_empty()) {
-            auto& mount = mounts.take_last();
-            MUST(mount.guest_fs().flush_writes());
-
-            auto mount_path = MUST(mount.absolute_path());
-            auto& mount_inode = mount.guest();
-            auto const result = VirtualFileSystem::the().unmount(mount_inode, mount_path->view());
-            if (result.is_error()) {
-                dbgln("Error during unmount of {}: {}", mount_path, result.error());
-                // FIXME: For unknown reasons the root FS stays busy even after everything else has shut down and was unmounted.
-                //        Until we find the underlying issue, allow an unclean shutdown here.
-                if (remaining_mounts <= 1)
-                    dbgln("BUG! One mount remaining; the root file system may not be unmountable at all. Shutting down anyways.");
-            } else {
-                unmount_was_successful = true;
-            }
-        }
-    }
+    if (auto result = VirtualFileSystem::the().unmount_all({}); result.is_error())
+        dmesgln("Unmounting all file systems failed due to {}", result.release_error());
```

Comment:
> even though it (probably?) fixes the current unmount bug we had for the root file system

I don't understand what you mean here. This problem has already been fixed in a previous PR (see #20665).

> Overall this reduces the resiliency to failing unmounts (and the ability to perform as much clean shutdown work as possible)

This is wrong. The previous code would also stop unmounting all filesystems if it failed to unmount a filesystem during that sequence. This change behaves exactly the same - it will stop unmounting filesystems on the first failure to unmount a filesystem. Whether this is wrong or not is up to a discussion and a fix in another future PR.

Reply:
I know for a fact that the old code would retry unmounting file systems, at least until we had only failing unmounts (there was some flag for “at least one successful unmount”). If this logic is no longer present, please leave a FIXME.

Reply:
I added a FIXME comment about trying again to unmount, in case of failure. The exact details are to be discussed later, for a future PR.